specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/examples/bookmarks/relations/GetAllBookmarkRelations.json

```diff
@@ -0,0 +1,33 @@
+{
+    "parameters": {
+      "api-version": "2019-01-01-preview",
+      "subscriptionId": "d0cfe6b2-9ac0-4464-9919-dccaee2e48c0",
+      "resourceGroupName": "myRg",
+      "workspaceName": "myWorkspace",
+      "operationalInsightsResourceProvider": "Microsoft.OperationalIinsights",
+      "bookmarkId": "2216d0e1-91e3-4902-89fd-d2df8c535096"
+    },
+    "responses": {
+      "200": {
+        "body": {
+          "value": [
+            {
+              "id": "/subscriptions/d0cfe6b2-9ac0-4464-9919-dccaee2e48c0/resourceGroups/myRg/providers/Microsoft.OperationalIinsights/workspaces/myWorkspace/providers/Microsoft.SecurityInsights/bookmarks/2216d0e1-91e3-4902-89fd-d2df8c535096/relations/4bb36b7b-26ff-4d1c-9cbe-0d8ab3da0014",
+              "name": "4bb36b7b-26ff-4d1c-9cbe-0d8ab3da0014",
+              "type": "Microsoft.SecurityInsights/cases/relations",
+              "kind": "CasesToBookmarks",
+              "etag": "190057d0-0000-0d00-0000-5c6f5adb0000",
+              "properties": {
+                "bookmarkId": "2216d0e1-91e3-4902-89fd-d2df8c535096",
+                "caseIdentifier": "afbd324f-6c48-459c-8710-8d1e1cd03812",
+                "caseTitle": "Test case",
+                "caseSeverity": "High",
+                "relationName": "4bb36b7b-26ff-4d1c-9cbe-0d8ab3da0014"
```

Comment:
>relationName [](start = 17, length = 12)

`name` and `relationName` appear redundant. Will they ever be different?

Reply:
All returned contracts have these common properties as ARM objects. name, type, kind. For bookmarks a name is a bookmark id, for relations name is the same as the relationName

Reply:
name, kind, are common properties shared by all contracts on this APIs. bookmarks also have a name property mapping to the bookmarkid, while the relations have a name property mapping to a relationName

Reply:
Discussed in person. There are cases where it will be different and it helps to disambiguate what the particular value is.

---
In reply to: [325297344](https://github.com/Azure/azure-rest-api-specs/pull/7191#discussion_r325297344) [](ancestors = 325297344)